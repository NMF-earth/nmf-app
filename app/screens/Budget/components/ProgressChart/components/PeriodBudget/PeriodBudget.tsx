import React from "react";
import { View } from "react-native";

import styles from "./PeriodBudget.styles";
import { Text } from "../../../../../../components";
import { t } from "../../../../../../utils";

interface Prop {
  period: string;
  periodEmissionsBudget: number;
}

const PeriodBudget = ({ periodEmissionsBudget = 0, period = "" }: Prop) => {
  let budget = periodEmissionsBudget.toString();
  let units = " kg";
  if (periodEmissionsBudget > 999) {
    budget = (periodEmissionsBudget / 1000).toFixed(2);
    units = " ton(s)";
  }

  return (
    <View style={styles.container}>
      <Text.Secondary bold center>
        {t("BUDGET_SCREEN_PROGRESS_CHART_LEGEND_PERIOD_BUDGET")}
        {period}
        {" : "}
        <Text.Secondary lightGray center>
          {budget + units}
        </Text.Secondary>
      </Text.Secondary>
    </View>
  );
};

export default PeriodBudget;
