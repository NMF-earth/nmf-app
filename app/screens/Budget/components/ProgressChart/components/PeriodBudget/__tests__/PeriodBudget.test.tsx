import React from "react";
import { create } from "react-test-renderer";

import PeriodBudget from "../PeriodBudget";

const props = {
  period: "this month",
  periodEmissionsBudget: 760,
};

jest.unmock("../PeriodBudget");

it("PeriodBudget renders correctly", () => {
  const tree = create(<PeriodBudget {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("PeriodBudget renders correctly if tons", () => {
  const tree = create(
    <PeriodBudget periodEmissionsBudget={14340} {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
