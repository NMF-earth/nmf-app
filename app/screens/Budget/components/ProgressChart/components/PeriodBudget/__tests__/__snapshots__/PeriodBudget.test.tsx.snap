// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeriodBudget renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "paddingBottom": 16,
      "paddingTop": 12,
    }
  }
>
  <Text.Secondary
    bold={true}
    center={true}
  >
    Budget for 
    this month
     : 
    <Text.Secondary
      center={true}
      lightGray={true}
    >
      <FormattedNumber
        maximumFractionDigits={0}
        value={760}
      />
       
      kg
    </Text.Secondary>
  </Text.Secondary>
</View>
`;

exports[`PeriodBudget renders correctly if tons 1`] = `
<View
  style={
    {
      "flex": 1,
      "paddingBottom": 16,
      "paddingTop": 12,
    }
  }
>
  <Text.Secondary
    bold={true}
    center={true}
  >
    Budget for 
    this month
     : 
    <Text.Secondary
      center={true}
      lightGray={true}
    >
      <FormattedNumber
        maximumFractionDigits={0}
        value={760}
      />
       
      kg
    </Text.Secondary>
  </Text.Secondary>
</View>
`;
